node/modules/lp2p/rcmgr.go

```diff
@@ -0,0 +1,188 @@
+package lp2p
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"go.uber.org/fx"
+
+	"github.com/libp2p/go-libp2p"
+	"github.com/libp2p/go-libp2p-core/network"
+	"github.com/libp2p/go-libp2p-core/peer"
+	"github.com/libp2p/go-libp2p-core/protocol"
+	rcmgr "github.com/libp2p/go-libp2p-resource-manager"
+
+	"github.com/application-research/estuary/metrics"
+	"github.com/filecoin-project/lotus/node/repo"
+
+	"go.opencensus.io/stats"
+	"go.opencensus.io/tag"
+)
+
+func NewDefaultLimiter() *rcmgr.BasicLimiter {
+	return rcmgr.NewDefaultLimiter()
+}
+
+func NewResourceManager(limiter *rcmgr.BasicLimiter) (network.ResourceManager, error) {
+	var opts []rcmgr.Option
+	libp2p.SetDefaultServiceLimits(limiter)
+	opts = append(opts, rcmgr.WithMetrics(rcmgrMetrics{}))
+	mgr, err := rcmgr.NewResourceManager(limiter, opts...)
+	if err != nil {
+		return nil, fmt.Errorf("error creating resource manager: %w", err)
+	}
+	return mgr, nil
+}
+
+func NewResourceManagerWithLifecycleRepo(lc fx.Lifecycle, repo repo.LockedRepo) (network.ResourceManager, error) {
```

Comment:
I think this function isnt used, right? we shouldnt have to add any dependencies on other lotus submodules

Reply:
It does yes. Removed it on the latest commit.

Reply:
I removed it on the latest commit.